Library/Homebrew/dependency_collector.rb

```diff
@@ -108,7 +108,8 @@ def parse_symbol_spec(spec, tags)
     case spec
     when :x11        then X11Requirement.new(spec.to_s, tags)
     when :xcode      then XcodeRequirement.new(tags)
-    when :macos      then MinimumMacOSRequirement.new(tags)
+    when :linux      then LinuxRequirement.new
+    when :macos      then tags.empty? ? MacOSRequirement.new : MinimumMacOSRequirement.new(tags)
```

Comment:
make `MacOSRequirement.new` handle the `tags.empty?` case instead of having the meaning overloaded.

Reply:
Done.